### PR TITLE
fs-477714-progressive-profiling-string-with-format

### PR DESCRIPTION
**Pull request by fangshi-okta**
## Description:

- if get display type as select and format is country-code, render the field as a dropdown with predefined list
- else render as a text box 

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:

Screen Shot 2022-03-01 at 1 13 17 PM

Screen Shot 2022-03-09 at 9 03 31 AM

### Reviewers:

@okta/ciamx 

### Issue:

- [OKTA-477714](https://oktainc.atlassian.net/browse/OKTA-477714)

### CODE REVIEW

File: playground/mocks/data/idp/idx/enroll-profile-new-additional-fields.json
Comment (justinabrokwah-okta) [inline playground/mocks/data/idp/idx/enroll-profile-new-additional-fields.json:66]: Just curious, why are we allowing text as an input for country-code?

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (vicentevillegas-okta) [inline src/v2/ion/ui-schema/ion-string-handler.js:63]: nit: I think the format wont be present for all attributes, wondering if we should make sure it is not empty, perhaps something like this:
`_.isNotEmpty(display.format)`

File: playground/mocks/data/idp/idx/enroll-profile-new-additional-fields.json
Comment (fangshi-okta) [reply playground/mocks/data/idp/idx/enroll-profile-new-additional-fields.json:66]: Good question. because we need to store admin's settings. every time after admin choose `text` display, next time, when admin comes back to the settings, the `text` option can be remembered. Previously, we do not add it as it is not needed in SIW and just an default value in ui. but for country code, it is one of the radio option in ui.

**Comment by fangshi-okta** [comment]
> Can you add a screenshot of you opening the dropdown?

screenshot added. thanks

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (fangshi-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js]: updated. Good call

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (pradeepdewda-okta) [inline src/v2/ion/ui-schema/ion-string-handler.js:63]: if it's a `select` why we need another `else if` here ?

File: playground/mocks/data/idp/idx/enroll-profile-new-additional-fields.json
Comment (justinabrokwah-okta) [reply playground/mocks/data/idp/idx/enroll-profile-new-additional-fields.json:66]: Oh okay, so it's another option we support then. Got it!

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (fangshi-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js]: updated. thanks

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (vicentevillegas-okta) [inline src/v2/ion/ui-schema/ion-string-handler.js:65]: nit: wondering if we can do something like this
image

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (pradeepdewda-okta) [inline src/v2/ion/ui-schema/ion-string-handler.js:64]: what's special about `countryUISchema` ? it still does have same `type` and `wide` attributes. The only change is `options` and `value`..

i think it's time we can eliminate `countryUISchema` and just update `options` and `value`

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (fangshi-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js:65]: good suggestion. maybe we can implement this when we decide to do the same pattern for timezone and language code. thanks!

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (fangshi-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js]: updated based on the suggestion. Thank you!

**Review by vicentevillegas-okta — APPROVED** [review]
I have tested locally okta-core + okta-ui + SIW, so these changes are working fine.

LGTM 👍 

I think you still need to get approval from a codeowner

Thank you @fangshi-okta for these changes!

File: src/v2/ion/ui-schema/ion-string-handler.js
Comment (pradeepdewda-okta) [inline src/v2/ion/ui-schema/ion-string-handler.js:58]: `display?.format === 'country-code'`

Comment (pradeepdewda-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js:58]: add `country-code` as constant instead of a string.. Add it to `RemediationConstants.js`

Comment (fangshi-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js]: Thanks for the suggestions! I think display value has been checked in `const display = ionField?.value?.value;` here I just want to check that `format` is not empty. country-code was added to RemediationConstants.js

Comment (pradeepdewda-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js]: can `format` be empty ?

Comment (fangshi-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js]: yeah only for `country-code` data type attribute, the format will contains value. any concerns?

Comment (pradeepdewda-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js]: want to confirm `display.format === ''` is possible ? if not then `display?.format === 'country-code'` should be enough right ?

Comment (fangshi-okta) [reply src/v2/ion/ui-schema/ion-string-handler.js]: updated based on our discussion. Thanks!

**Review by pradeepdewda-okta — APPROVED** [review]

**Comment by pradeepdewda-okta** [comment]
Add a unit test and then merge.

**Pull request closed without merging**
